R/utils-get_code_dependency.R

```diff
@@ -33,18 +33,11 @@ get_code_dependency <- function(code, names, check_names = TRUE) {
     return(code)
   }
 
-  # If code is bound in curly brackets, remove them.
-  # TODO: rethink if this is still needed when code is divided by calls?
-  tcode <- trimws(code)
-  if (any(grepl("^\\{.*\\}$", tcode))) {
-    tcode <- sub("^\\{(.*)\\}$", "\\1", tcode)
-  }
-
-  parsed_code <- parse(text = tcode, keep.source = TRUE)
+  parsed_code <- parse(text = trimws(code), keep.source = TRUE)
 
   pd <- utils::getParseData(parsed_code)
   pd <- normalize_pd(pd)
-  calls_pd <- extract_calls(pd)
+  calls_pd <- extract_calls(pd) # STILL NEEDED for check_names
```

Comment:
It doesn't really look needed by `check_names`. IN the `if` below there are `unlist` anyway which means that instead of loop you can browse `pd` directly
```r
# instead
symbols <- unlist(lapply(calls_pd, function(call) call[call$token == "SYMBOL", "text"]))

# do
symbols <- pd[pd$token == "SYMBOL", "text"]
```

Also lines 36-39 should be moved inside of `if (check_names) {...}`

Reply:
Thanks, this simplified the code a bit. Moved it in this commit. 4ea2ae2be86b08bed72b41dd390f33c418bf4275

Do you think we could create a specific function for this check and move it out of `get_code_dependency` so it's shorter?